glue/core/tests/test_edit_subset_mode.py

```diff
@@ -87,3 +94,40 @@ def test_combines_make_copy(self):
         self.edit_mode.edit_subset = self.data.new_subset()
         self.edit_mode.update(self.data, self.state2)
         assert self.edit_mode.edit_subset.subset_state is not self.state2
+
+
+def test_no_double_messaging():
+
+    # Make sure that when we create a new subset via EditSubsetMode, we don't
+    # get two separate messages for creation and updating, but instead just a
+    # single create message with the right subset state.
+
+    handler = MagicMock()
+
+    app = Application()
+
+    class Listener(HubListener):
+        pass
+
+    listener = Listener()
+
+    app.session.hub.subscribe(listener, SubsetCreateMessage, handler=handler)
+    app.session.hub.subscribe(listener, SubsetUpdateMessage, handler=handler)
+
+    data = Data(x=[1, 2, 3, 4, 5])
+
+    app.data_collection.append(data)
+
+    cmd = ApplySubsetState(data_collection=app.data_collection,
+                           subset_state=data.id['x'] >= 3,
+                           override_mode=ReplaceMode)
+
+    app.session.command_stack.do(cmd)
+
+    assert handler.call_count == 1
+    message = handler.call_args[0][0]
+    assert isinstance(message, SubsetCreateMessage)
+    assert isinstance(message.subset.subset_state, InequalitySubsetState)
+    assert message.subset.subset_state.left is data.id['x']
+    assert message.subset.subset_state.operator is operator.ge
+    assert message.subset.subset_state.right == 3
```

Comment:
```suggestion
    assert message.subset.subset_state.right == 3


def test_broadcast_to_collection():
    """A data collection input works on each data component"""

    handler = MagicMock()

    app = Application()

    class Listener(HubListener):
        pass

    listener = Listener()

    app.session.hub.subscribe(listener, SubsetCreateMessage, handler=handler)
    app.session.hub.subscribe(listener, SubsetUpdateMessage, handler=handler)

    data = Data(x=[1, 2, 3, 4, 5])
    data2 = Data(x=[2, 3, 4, 5, 6])

    app.data_collection.append(data)
    app.data_collection.append(data2)

    cmd = ApplySubsetState(data_collection=app.data_collection,
                           subset_state=data.id['x'] >= 3,
                           override_mode=ReplaceMode)

    app.session.command_stack.do(cmd)

    assert len(data2.subsets) == 1
    assert data2.subsets[0].label == 'Subset 1'
    # fails with `IncompatibleAttribute` exception
    # assert data2.get_subset_object(subset_id='Subset 1', cls=NDDataArray)

    assert handler.call_count == 2

    assert data2.subsets[0].subset_state.left is data.id['x']
    assert data2.subsets[0].subset_state.operator is operator.ge
    assert data2.subsets[0].subset_state.right == 3
```
basic test to check that subset is broadcast to a second component in the dataset.